src/app/shared/services/layout.service.ts

```diff
@@ -26,25 +32,26 @@ export class LayoutService {
    * Disabled for instance for displaying a task, as the task iframe is set to fill the screen to the bottom */
   contentFooter$ = this.contentFooter.asObservable();
 
-  private initialized = false;
+  private configured = false;
 
   /**
    * This method allows to defer layout initialization to any consumer, expectedly routes.
    * Only first call in taken into account, later calls are ignored.
-   * @param expanded initial expanded value
-   * @param canToggle Defines for the lifetime of the app if the left menu can be toggled or not
-   * @param showTopRightControls Defines for the lifetime of the app if the top right controls are shown
+   * @param {object} options
+   * @param options.expanded initial expanded value
+   * @param options.canToggle Defines for the lifetime of the app if the left menu can be toggled or not
+   * @param options.showTopRightControls Defines for the lifetime of the app if the top right controls are shown
    */
-  initialize(expanded: boolean, canToggle: boolean, showTopRightControls: boolean): void {
-    if (this.initialized) return;
+  configure({ expanded, showTopRightControls = true, canToggle = true }: ConfigureOptions): void {
```

Comment:
For one-time configuration: It is as this would need to be set only once, but at the end, shouldn't the left-menu display/hide because of visiting a chapter/task (when not using the left menu) use the same mecanism?
Look strange to me that item-by-id deals with roughly the same thing (setting the "expanded") twice but by different ways.

BTW, there is maybe no reason to return when already configured. We could consider that "canToggle" and "showTopRightControls" may be changed. As there are no need for that for now, you don't need to change that, but if it may help with my remark above... why not.

Reply:
> For one-time configuration: It is as this would need to be set only once, but at the end, shouldn't the left-menu display/hide because of visiting a chapter/task (when not using the left menu) use the same mecanism?

It is meant to be set only at app startup because it's the only case when there is no animation, that's why it was originally called `initialize()`.
After configuration, the layout can be toggled using the other method `toggleFullFrameContent` which 
has intentionally less parameters and enables animations.
In my opinion, the 2 cases at-startup+no animation/after-startup+animations obey to different use cases, thus they are separate functions.

> Look strange to me that item-by-id deals with roughly the same thing (setting the "expanded") twice but by different ways.

Fixed, only the solve missing path error is only handled.

Reply:
> It is meant to be set only at app startup because it's the only case when there is no animation, that's why it was originally called `initialize()`. After configuration, the layout can be toggled using the other method `toggleFullFrameContent` which has intentionally less parameters and enables animations. 

The animation could easily be solved by replacing `animated: false` by `animated: alreadyConfigured` :-)

And between `this.layoutService.configure({ expanded: false })` and `this.layoutService.toggleFullFrameContent(false)`, there is the same number of arg... and 1 more character 😁

Ok, that is because I haven't check the different in usage between the two calls but I can check it...

### First, in layout service: 
L53 is useless as `canToggle` is `false` by L52.
On L52, `this.fullFrameContent.value.canToggle` can only be false if `!this.configured` so one of the 2 is useless as well. 

So you end up with:
```ts
toggleFullFrameContent(expanded: boolean): void {
    if (!this.configured) return;
    this.fullFrameContent.next({ expanded, canToggle: false, animated: true });
  }
```

### Calls, in item-by-id
1)
```ts
this.layoutService.configure({ expanded: isTask(data.item) });
```
called on `itemDataSource.state$` change, when ready, the first time

2)
```ts
this.layoutService.toggleFullFrameContent(isTask(item) && !(history.state as Record<string, boolean | undefined>).preventFullFrame);
```
called on  `itemDataSource.state$` change, when ready, each time (including the first time I think).

I suppose the first time it is called twice. 
A) If (2) is called before (1), it will be ignored (because of L52)... so the history prevent full frame is ignored, is it ok? (possibly a bug ?)
B) If (1) is called before (2) is (1) is just useless, right?

I suppose that the order of definition is deterministic, but quite unknown to me (via command order?). Possibly risky if someone reorganize the file.
Anyway, if I haven't done any mistake in my analysis, the fact that

> item-by-id deals with roughly the same thing (setting the "expanded") twice but by different ways

(from my previous review) is not really a detail as it leads to either a useless code or a possible bug... and even if there is no problem, it is unclear enough not to treat them separately (except if I misunderstood a major aspect).

Reply:
For the useless line in layout service: fixed
---
For calls in item by id:
A) Call (1) will always be called before call (2) because the subscription triggering call (1) is set before subscription triggering call (2). Anyway the first time, `history.state` has no chance to be populated since the history state can only be populated via an internal navigation. No possible bug here to me.
B) If (1) is called before (2), then (2) is indeed effectless. This is the case only when loading the first app route, afterwards, the call (1) will always be ignored and call (2) will be the one taken into account.

Whatever the calls order, the behavior will remain the same:
1st load: the special case
- (1) called before (2): layout is configured by (1), (2) is effectless because `fullFrame.active` is the same as current one.
- (2) called before (1): (2) is ignored because layout isn't configured yet, (1) configures the layout.
nth load: only via in-app navigation
- (1) called before (2): (1) is ignored because layout is configured, (2) is taken into account
- (2) called before (1): (2) is taken into account, (1) is ignored because layout is configured.

To me, it seems deterministic and resilient to changes, doesn't it?

Reply:
I forgot to address this comment:
> The animation could easily be solved by replacing animated: false by animated: alreadyConfigured :-)
And between this.layoutService.configure({ expanded: false }) and this.layoutService.toggleFullFrameContent(false), there is the same number of arg... and 1 more character :grin:

I can replace `animated: false` by `animated: alreadyConfigured`, but then it still means I have to _configure_ something, causing the second comment (suggesting to merge the 2 methods) to be irrelevant, unless I'm missing something.
Anyway the 2 methods do not have the same number of arguments, `configure()` has 3 arguments with 2 having defaults values while `toggleFullFrame()` has only one.

Reply:
Maybe it will be easier to discuss this over the phone